ScienceDirect.js

```diff
@@ -48,26 +51,51 @@ function detectWeb(doc, url) {
 	}
 }
 
-function getPDFLink(doc) {
-	var pdfLink = ZU.xpathText(doc, '//div[@id="articleToolbar"]//a[@id="pdfLink" and not(@title="Purchase PDF")]/@href');
-	if (!pdfLink) {
-		pdfLink = ZU.xpathText(doc, '//div[@class="extendedPdfBox"]//a[@id="pdfLink" and not(@title="Purchase PDF")]/@href');
+function getPDFLink(doc, onDone) {
+	// No PDF access ("Get Full Text Elsewhere" or "Check for this article elsewhere")
+	if (doc.querySelector('.accessContent') || doc.querySelector('.access-options-link-text')) {
+		Zotero.debug("PDF is not available");
+		onDone();
+		return;
 	}
-	if (!pdfLink) {
-		pdfLink = ZU.xpathText(doc, '//div[@class="PdfEmbed"]/object/@data');
+	
+	// Some pages still have the PDF link available
+	var pdfURL = attr(doc, '#pdfLink', 'href');
+	if (pdfURL) {
+		onDone(pdfURL);
+		return;
 	}
-	if (!pdfLink) {
-		var mainPdf = ZU.xpath(doc, '//div[@id="articleToolbar" or @class="Toolbar"]//a[contains(@href, "main.pdf")]');
-		//we only look at the first match
-		if (mainPdf.length>0) {
-			var classes = mainPdf[0].class || '';
-			//Z.debug(classes);
-			if (mainPdf[0].title!="Purchase PDF" && classes.indexOf('excerptLink')==-1 && classes.indexOf('purchaseSprite')==-1 && classes.indexOf('ref-article-pdf')==-1 && classes.indexOf('cLink')==-1) {
-				pdfLink = mainPdf[0].href;
-			}
+	
+	// For pages that use the new JS drop-down, the DOM doesn't seem to have the intermediate page URL
+	// after the page load, but it's in the initial HTML, so fetch the page again. This is awful, and
+	// we should try to find a better way of getting the URL (which is obviously available to JS in
+	// the page).
```

Comment:
OK, that page works (and maybe all "Download full text in PDF" + menu pages). I was testing on the (newer?) "Download PDF" + menu pages, like http://www.sciencedirect.com/science/article/pii/S0006349512000835, which don't appear to have the link until you click on the menu.

I've updated it to get the immediate URL directly for the .PdfEmbed case. If you have an example of the other we can add support for that too (but we can probably push this out without that, since the only consequence is slower saving).